plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2399,7 +2399,9 @@ private void createVif(final LibvirtVMDef vm, final NicTO nic, final String nicA
             }
         }
 
-        vm.getDevices().addDevice(getVifDriver(nic.getType(), nic.getName()).plug(nic, vm.getPlatformEmulator(), nicAdapter));
+        InterfaceDef device = getVifDriver(nic.getType(), nic.getName()).plug(nic, vm.getPlatformEmulator(), nicAdapter);
+        device.setQueues(queues);
```

Comment:
thoughts on adding: if (nic.getType() == 'virtio' && queues > 1) {

Reply:
We can, but it is duplicating libvirt logic to some degree. If libvirt/qemu ever add multi queue processing to other nic drivers we would need to adjust this. If we leave the check out libvirt will just ignore it if it doesn't apply since it is the authority. If you do want to add a check I would rather add it to LibvirtVMDef.java since all the config logic lives there already.

In looking at this I did find another place it is required to support the nic hot plug logic. So I'll get it added there as well.

Reply:
Thanks Nate, agreed, LibvirtVMDef.java would be where we should put it.